### PR TITLE
Support integer64

### PR DESCRIPTION
**Pull request by ngfreiter**
This addresses [CIMPL-622](https://standardhealthrecord.atlassian.net/browse/CIMPL-622) adds support for the [integer64 type](http://build.fhir.org/datatypes.html#integer64) introduced in R5. The only place I found that needed updating was assignment rules. Note that integer64 is treated differently from other numeric types, because we serialize it as a string in JSON, not as a number.

To test this, Extension works well. Here is an example I used to test out various different rules:
```
Extension: MyExtension
* value[x] only integer64
* valueInteger64 MS
* valueInteger64 1..1
* valueInteger64 = 1234567890
```

### CODE REVIEW

**Review by cmoesel — CHANGES_REQUESTED** [review]
The FHIR spec says long has the range `-9,223,372,036,854,775,808` to `9,223,372,036,854,775,807`.  I tried this:
```
* valueInteger64 = 9223372036854775807
```
and ended up w/ JSON having this:
```
"patternInteger64": "9223372036854776000",
```

I expect that somewhere in here, it's getting converted to a JavaScript Number, which is a floating-point value, so can only reliably represent whole numbers up to `9007199254740991` (represented by `Number.MAX_SAFE_INTEGER`).  After that, they get rounded (as we see above).

We'll probably need to either keep the value as a `string` the whole way through the process or employ other number classes (like `BigInt`).

I also had one question.  I noticed that we do not allow the user to specify the integer64 as a string in the FSH:
```
* valueInteger64 = "1234567890"
```
This fails with: "Cannot assign string value: 1234567890. Value does not match element type: integer64".  Since FHIR says to represent this as a string, I wonder if it might be more user-friendly to also allow for that in FSH?

I don't remember having this conversation about dates, but I noticed that I can say:
```
* valueDateTime = 2020-01-14T09:00:00.0Z
```
_OR_
```
* valueDateTime = "2020-01-14T09:00:00.0Z"
```
and SUSHI happily accepts them both.  This seems like a very similar situation as integer64.

**Comment by ngfreiter** [comment]
@cmoesel your comments should be addressed https://github.com/FHIR/sushi/pull/728/commits/5176aba98c574681c7f3c65383cedd6ad8303230 and https://github.com/FHIR/sushi/pull/728/commits/223e1a8cab595ecaafcb6bbae9414865c40fb46b

File: test/fhirtypes/ElementDefinition.assignNumber.test.ts [hunk 1]
```diff
@@ -452,4 +454,95 @@ describe('ElementDefinition', () => {
       expect(valueX.fixedInteger).toBeUndefined();
     });
   });
+
```
[the remark this answers is not in the document]
Comment (cmoesel) [reply test/fhirtypes/ElementDefinition.assignNumber.test.ts:457]: And after some manual testing, it seems that, in fact, the `BigInt` _will_ break the normal integer use cases.  The following FSH causes an error in SUSHI:

```
Extension: MyNormalIntExtension
* value[x] only integer
* valueInteger MS
* valueInteger 1..1
* valueInteger = 12345
```

The error is:
> error SUSHI encountered the following unexpected error: Do not know how to serialize a BigInt

This happens if you try to assign a normal integer in an instance as well.

**Review by cmoesel — CHANGES_REQUESTED** [review]
This fixes the issue w/ `integer64`, but now that you are storing all integers as `BigInt` values, it seems that some of the traditional integer handling is broken.  See individual comments.

File: test/fhirtypes/ElementDefinition.assignNumber.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ngfreiter) [reply test/fhirtypes/ElementDefinition.assignNumber.test.ts:457]: Oops, thanks for pointing this out. Fixed in https://github.com/FHIR/sushi/pull/728/commits/109e3520bb6e8a665aa600ec61930275dc3c06fc.

File: test/fhirtypes/ElementDefinition.assignNumber.test.ts [hunk 2]
```diff
@@ -452,4 +454,95 @@ describe('ElementDefinition', () => {
...
+      expect(valueInteger64.patternInteger64).toBeUndefined();
+      expect(valueInteger64.fixedInteger64).toBe('123');
+    });
+
+    it('should assign a large integer to an integer64 without losing precision', () => {
+      valueInteger64.assignValue(BigInt('12345678901234567890'));
+      expect(valueInteger64.patternInteger64).toBe('12345678901234567890');
+      expect(valueInteger64.fixedInteger64).toBeUndefined();
+    });
+
```
[the remark this answers is not in the document]
Comment (ngfreiter) [reply test/fhirtypes/ElementDefinition.assignNumber.test.ts:504]: Added comments in https://github.com/FHIR/sushi/pull/728/commits/109e3520bb6e8a665aa600ec61930275dc3c06fc.

**Review by cmoesel — APPROVED** [review]
Yeah, OK.  This is definitely something I can get behind!

**Review by julianxcarter — APPROVED** [review]
Everything looks good and works well!

**Pull request merged**